src/backend/utils/adt/pipelinefuncs.c

```diff
@@ -1012,3 +1011,57 @@ pipeline_transforms(PG_FUNCTION_ARGS)
 
 	SRF_RETURN_DONE(funcctx);
 }
+
+static void
+ipc_queue_wait_till_empty(ipc_queue *ipcq)
+{
+	while (true)
+	{
+		if (ipc_queue_is_empty(ipcq))
+			break;
+		pg_usleep(10);
```

Comment:
This isn't going to burn a *ton* of cycles but the sleep period can definitely be increased here by several orders of magnitude with no noticeable usability difference. Unless there's a use case where 10 μs of resolution matters we might as well save the cycles.